src/Reflector/Generic.php

```diff
@@ -0,0 +1,190 @@
+     */
+    private function reflectNode(Node $node, Node\Stmt\Namespace_ $namespace = null, $filename = null)
+    {
+        if ($node instanceof Node\Stmt\Class_) {
+            return ReflectionClass::createFromNode(
+                $node,
+                $namespace,
+                $filename
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * Process and reflect all the matching identifiers found inside a namespace node
+     *
+     * @param Node\Stmt\Namespace_ $namespace
+     * @param Identifier $identifier
+     * @param string|null $filename
+     * @return Reflection[]
+     */
+    private function reflectFromNamespace(
+        Node\Stmt\Namespace_ $namespace,
+        Identifier $identifier,
+        $filename
+    ) {
+        $reflections = [];
+        foreach ($namespace->stmts as $node) {
+            $reflection = $this->reflectNode($node, $namespace, $filename);
+
+            if (null !== $reflection && $identifier->getType()->isMatchingReflector($reflection)) {
+                $reflections[] = $reflection;
+            }
+
+        }
+        return $reflections;
+    }
+
+    /**
+     * Reflect identifiers from an AST. If a namespace is found, also load all the
+     * matching identifiers found in the namespace
+     *
+     * @param Node[] $ast
+     * @param string|null $filename
+     * @param Identifier $identifier
+     * @return Reflection[]
+     */
+    private function reflectFromTree(array $ast, $filename, Identifier $identifier)
+    {
+        $reflections = [];
+        foreach ($ast as $node) {
+            if ($node instanceof Node\Stmt\Namespace_) {
+                $reflections = array_merge(
+                    $reflections,
+                    $this->reflectFromNamespace($node, $identifier, $filename)
+                );
+            } else if ($node instanceof Node\Stmt\Class_) {
+                $reflection = $this->reflectNode($node, null, $filename);
+                if ($identifier->getType()->isMatchingReflector($reflection)) {
+                    $reflections[] = $reflection;
+                }
+            }
+        }
+        return $reflections;
+    }
+
+    /**
+     * Get an array of reflections found in a LocatedSource
+     *
+     * @param LocatedSource $locatedSource
+     * @param Identifier $identifier
+     * @return Reflection[]
+     */
+    private function getReflections(LocatedSource $locatedSource, Identifier $identifier)
+    {
+        return $this->reflectFromTree(
+            (new Parser(new Lexer))->parse($locatedSource->getSource()),
+            $locatedSource->getFileName(),
+            $identifier
+        );
+    }
+
+    /**
+     * Get all identifiers of a matching identifier type from a file
+     *
+     * @param IdentifierType $identifierType
+     * @return Reflection[]
+     */
+    public function getAllByIdentifierType(IdentifierType $identifierType)
+    {
+        $identifier = new Identifier('*', $identifierType);
```

Comment:
@Ocramius I've created issue #21 to deal with this janky identifier stuff in `getAllByIdentifierType`